addons/binding/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/handler/HarmonyHubHandler.java

```diff
@@ -76,6 +80,8 @@
 
     private static final int HEARTBEAT_INTERVAL = 30;
 
+    private ScheduledExecutorService buttonExecutor = Executors.newSingleThreadScheduledExecutor();
```

Comment:
I think you want to use a single thread executor so the order of commands stays the same?